packages/iframe-execution-environment-service/src/IframeExecutionEnvironmentService.ts

```diff
@@ -110,9 +110,11 @@ export class IframeExecutionEnvironmentService
   }
 
   public async terminatePlugin(pluginName: string) {
-    const jobId = this.pluginToJobMap.get(pluginName);
-    jobId && this.terminate(jobId);
-    this._removePluginHooks(pluginName);
```

Comment:
removePluginHooks was moved to `_removePluginAndJobMapping` to get handled in the lifecycle regardless if `terminate` or `terminatePlugin` is called